### PR TITLE
fix type assertion

### PR DESCRIPTION
**Pull request by sivchari**
**Please provide a description of this PR:**
Fix to do type assertion.

### CODE REVIEW

File: pilot/pkg/config/file/store.go [hunk 1]
```diff
@@ -364,7 +364,10 @@ func (s *KubeSource) parseChunk(r *collection.Schemas, name string, lineNum int,
 	if err != nil {
 		return kubeResource{}, fmt.Errorf("failed parsing JSON for built-in type: %v", err)
 	}
-	objMeta := obj.(metav1.Object)
+	objMeta, ok := obj.(metav1.Object)
+	if !ok {
+		return kubeResource{}, errors.New("failed to type assertion")
```
Comment (bianpengyuan) [inline pilot/pkg/config/file/store.go:369]:
```suggestion
		return kubeResource{}, errors.New("failed to assert type of object metadata")
```

**Review by bianpengyuan — COMMENTED** [review]
/ok-to-test

File: pilot/pkg/config/file/store.go (hunk 1 above)
Comment (sivchari) [reply pilot/pkg/config/file/store.go]: I fixed it.

File: pilot/.golangci.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+linters-settings:
```
Comment (bianpengyuan) [inline pilot/.golangci.yml:1]: accidental check in?

Comment (sivchari) [reply pilot/.golangci.yml]: Oh, sorry. I fixed it.

**Review by bianpengyuan — APPROVED** [review]

**Pull request merged**